File: in_Python/1313 Decompress Run-Length Encoded List.py
class Solution(object):
    def decompressRLElist(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = []
        for i in range(1,len(nums),2):
            a = nums[i-1]
            b = nums[i]
            while a != 0:
                res.append(b)
                a -= 1
        return res

File: in_Java/1313 Decompress Run-Length Encoded List.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        List<Integer> l = new ArrayList<>();
        for(int i = 0; i < nums.length-1; ){
            int a = nums[i];
            int b = nums[i+1];
            while(a-- != 0){
                l.add(b);
            }
            i += 2;
        }
        int[] res = new int[l.size()];
        for(int i = 0; i < res.length; i++){
            res[i] = l.get(i);
        }
        return res;
    }
}